aws_sagemaker_studio/sagemaker_studio_image_build/tensorflow_bring_your_own/tensorflow_bring_your_own.ipynb

```diff
@@ -0,0 +1,740 @@
+    "\n",
+    "## Permissions\n",
+    "\n",
+    "Running this notebook requires permissions in addition to the normal `SageMakerFullAccess` execution role permissions. This is because it: \n",
+    "\n",
+    "1. Creates a new repository and pushes built images to [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/)\n",
+    "2. Utilizes [AWS Code Build](https://aws.amazon.com/codebuild/) to build new docker images \n",
+    "\n",
+    "\n",
+    "## The example\n",
+    "\n",
+    "In this example we show how to package a custom TensorFlow container with Amazon SageMaker studio with a Python example which works with the CIFAR-10 dataset and uses TensorFlow Serving for inference. However, different inference solutions other than TensorFlow Serving can be used by modifying the docker container.\n",
+    "\n",
+    "In this example, we use a single image to support training and hosting. This simplifies the procedure because we only need to manage one image for both tasks. Sometimes you may want separate images for training and hosting because they have different requirements. In this case, separate the parts discussed below into separate Dockerfiles and build two images. Choosing whether to use a single image or two images is a matter of what is most convenient for you to develop and manage.\n",
+    "\n",
+    "If you're only using Amazon SageMaker for training or hosting, but not both, only the functionality used needs to be built into your container.\n",
+    "\n",
+    "[CIFAR-10]: http://www.cs.toronto.edu/~kriz/cifar.html\n",
+    "\n",
+    "## The workflow\n",
+    "\n",
+    "This notebook is divided into two parts: _building_ the container and _using_ the container."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Part 1: Packaging and Uploading your Algorithm for use with Amazon SageMaker\n",
+    "\n",
+    "### An overview of Docker\n",
+    "\n",
+    "If you're familiar with Docker already, you can skip ahead to the next section.\n",
+    "\n",
+    "For many data scientists, Docker containers are a new technology. But they are not difficult and can significantly simply the deployment of your software packages. \n",
+    "\n",
+    "Docker provides a simple way to package arbitrary code into an _image_ that is totally self-contained. Once you have an image, you can use Docker to run a _container_ based on that image. Running a container is just like running a program on the machine except that the container creates a fully self-contained environment for the program to run. Containers are isolated from each other and from the host environment, so the way your program is set up is the way it runs, no matter where you run it.\n",
+    "\n",
+    "Docker is more powerful than environment managers like conda or virtualenv because (a) it is completely language independent and (b) it comprises your whole operating environment, including startup commands, and environment variable.\n",
```

Comment:
```suggestion
    "Docker is more powerful than environment managers like conda or virtualenv because (a) it is completely language independent and (b) it comprises your whole operating environment, including startup commands, and environment variables.\n",
```

Reply:
Updated